### ISSUE TITLE
can't scroll through tag filter screen with keyboard when not enough tasks are shown

### ISSUE THREAD

**Issue by Alys** [HabitRPG/habitica#12613]
If you have a lot of tags, then when you select one that has no tasks or only a few tasks assigned to it, you cannot use the keyboard arrow keys to scroll all the way down the tag filter panel. You can still use the mouse/trackpad to scroll.

Reported by @shewhoknits (e1d43f13-4a7b-4188-8844-127b4f63c67c), reproduced by me. Firefox on Ubuntu for me.

Steps to reproduce:
1. Give yourself very many tags. I'll paste my set of test tags into a separate comment here. It exactly mimics shewhoknits's tags except with the names changed for privacy. I don't know if that many are needed, but a small set of tags does not allow the bug to be reproduced.
1. Reload the website to ensure it has picked up the latest tag data for your account.
1. Go into the tag filter panel and select one tag. Ensure that it is a tag that does not have any tasks assigned. (The same bug occurs if you have only a small number of tasks assigned to it but not if you have a lot of tasks.)
1. Reopen the tag panel if it closed itself (e.g., if you accidentally moved your mouse too much).
1. Attempt to scroll down through the tag filter panel using the arrow keys on your keyboard. Observe that you can't scroll all the way down.
1. Attempt to scroll down through the tag filter panel using your mouse or trackpad. Observe that you can scroll all the way down.

The desired behaviour is that scrolling with the arrow keys should allow you to get to the bottom of your tags list, just like the mouse does. (This is useful for, for example, people with manual dexterity problems who have trouble using a mouse.)

_I've labelled this as "medium level coding" but that's a bit of a guess. If you're not reasonably confident you could work out how to solve this, you may want to pick a different issue._

**Comment by Alys** [comment]
_I've labelled this as "medium level coding" but that's a bit of a guess. If you're not reasonably confident you could work out how to solve this, you may want to pick a different issue._

**Comment by Alys** [comment]
Sample tag data that allows the bug to be reproduced:

```
    "tags": [
        {
            "id": "cbf2a132-4b44-4ea1-85f3-1f94f11627eb",
            "name": "test",
        },
        {
            "id": "5514226c-4351-4932-b038-2dc44fd57ebd",
            "name": "test",
        },
        {
            "id": "0f009c14-26ad-4135-9b59-05257c858316",
            "name": "test",
        },
        {
            "id": "8f7540b3-0c70-488a-a64a-30cb28e1ff12",
            "name": "test",
        },
        {
            "id": "6f8a221a-68a3-4a39-ac8c-a8d18eef7344",
            "name": "test",
        },
        {
            "id": "b776234d-b499-4449-9fa6-6a6f87b166b0",
            "name": "test",
            "challenge": true
        },
        {
            "id": "61901f6c-5846-415f-896d-b4fc94ddcff1",
            "name": "test",
            "challenge": true
        },
        {
            "id": "b59ef4c9-ae70-4a24-946a-dbf346664181",
            "name": "test",
            "challenge": true
        },
        {
            "id": "61f44e99-eed0-4472-879c-5e3e957d121a",
            "name": "test",
            "challenge": true
        },
        {
            "id": "46aac13a-ab1f-45c2-a60d-5678c6edb0c7",
            "name": "test",
            "challenge": false
        },
        {
            "id": "a386a38d-14a2-47fa-806b-07d7859c9026",
            "name": "test",
            "challenge": false
        },
        {
            "id": "4da32dc5-930e-4033-9fe2-fd4d4fbe8d58",
            "name": "test",
            "challenge": true
        },
        {
            "id": "5bd251a6-3b9f-4231-b19a-6763624fc28a",
            "name": "test",
            "challenge": true
        },
        {
            "id": "f1af7268-ef3d-46b9-9899-b97cc283c43d",
            "name": "test",
            "challenge": true
        },
        {
            "id": "d384c280-a2b9-4e9b-b022-1c4ea2b12fd2",
            "name": "test",
            "challenge": true
        },
        {
            "id": "d6127328-855b-400c-947a-bb9676a63eda",
            "name": "test",
            "challenge": true
        },
        {
            "id": "6cca20d6-ba65-40a4-aed8-130c140a4bdd",
            "name": "test",
            "challenge": true
        },
        {
            "id": "2b097089-9b6a-4c78-a1e0-a2d3ff174d33",
            "name": "test",
            "challenge": false
        },
        {
            "id": "0a2159d9-b81f-408a-9758-68d01a71fe4c",
            "name": "test",
            "challenge": true
        },
        {
            "id": "8afa3edf-f02c-4a95-b7ce-8ad386004d93",
            "name": "test",
            "challenge": true
        },
        {
            "id": "9d8c92d1-546d-41ba-8fc4-b9a5156b1c43",
            "name": "test",
            "challenge": true
        },
        {
            "id": "937a5bdf-ce1a-4362-8b82-a3ea842636c0",
            "name": "test",
            "challenge": true
        },
        {
            "id": "64675c44-8657-4404-abe5-b4b7342051ef",
            "name": "test",
            "challenge": true
        },
        {
            "id": "a438f9b7-4b51-4d36-9a06-1304bd08512a",
            "name": "test",
            "challenge": true
        },
        {
            "id": "09cd1f55-7b46-4e4e-b293-3f192e196420",
            "name": "test",
            "challenge": true
        },
        {
            "id": "93e60303-6c94-4309-ad6c-54ebc8742450",
            "name": "test",
            "challenge": true
        },
        {
            "id": "22e86ffd-b892-467c-86ac-e67f54ccfa1b",
            "name": "test",
            "challenge": true
        },
        {
            "id": "1c2ad9ad-cf5a-489f-a3ad-bbd20c3e6627",
            "name": "test",
            "challenge": true
        },
        {
            "id": "67637896-16a2-4bbc-836c-492560bb01ec",
            "name": "test",
            "challenge": true
        },
        {
            "id": "e91d8377-8f9d-4b8a-95b3-2e29c2a4c332",
            "name": "test",
            "challenge": true
        },
        {
            "id": "432f8868-779d-4fdc-823d-f858eb9e2a5d",
            "name": "test",
            "challenge": true
        },
        {
            "id": "41942956-58d3-4ee8-8bc1-207d456fc0c7",
            "name": "test",
            "challenge": true
        },
        {
            "id": "6f41654b-a6c3-4d6f-9f84-34a9acf7018e",
            "name": "test",
            "challenge": true
        },
        {
            "id": "90411174-968b-4a0b-a398-c8873e4af8c3",
            "name": "test",
            "challenge": true
        },
        {
            "id": "0d91eb22-463e-4ed2-8efd-1f30ab22ac84",
            "name": "test",
            "challenge": true
        },
        {
            "id": "84b27838-2d0c-4bc0-8eb7-a7b64ac2f192",
            "name": "test",
            "challenge": true
        },
        {
            "id": "c5876cfa-1a3a-4909-858e-c03ae07f3736",
            "name": "test",
            "challenge": true
        },
        {
            "id": "78e426cf-bbb7-4122-89b7-391587bb2896",
            "name": "test",
            "challenge": true
        }
    ],
```

**Comment by hazwong** [comment]
Could I try to work on this? I'm thinking we can add a scroll bar to the tag filter panel itself, so that when the user adds a lot of tags, the tag panel doesn't extend beyond the page.

**Comment by Alys** [comment]
@hazwong Thank you for taking this on! I've marked it as in progress. Sorry we didn't reply sooner!

**Comment by paglias** [comment]
I was looking at @hazwong 's PR and actually I'm not able to replicate the issue on Safari, it scrolls correctly using the keyboard arrows, @Alys is it still happening for you?

**Comment by Alys** [comment]
The bug no longer happens for me.
